core/amber/src/main/scala/edu/uci/ics/amber/clustering/ClusterListener.scala

```diff
@@ -47,28 +47,42 @@ class ClusterListener extends Actor with AmberLogging {
       .map(_.address)
   }
 
+  private def forcefullyStop(jobService: WorkflowJobService, cause: Throwable): Unit = {
+    jobService.client.shutdown()
+    jobService.stateStore.statsStore.updateState(stats =>
+      stats.withEndTimeStamp(System.currentTimeMillis())
+    )
+    jobService.stateStore.jobMetadataStore.updateState { jobInfo =>
```

Comment:
almost all the logic is regarding the jobService. maybe consider write this function as `JobService.forcefullyStop`?